source/fuzz/transformation_composite_insert.cpp

```diff
@@ -219,5 +173,65 @@ std::unordered_set<uint32_t> TransformationCompositeInsert::GetFreshIds()
   return {message_.fresh_id()};
 }
 
+void TransformationCompositeInsert::AddDataSynonymFacts(
+    opt::IRContext* ir_context,
+    TransformationContext* transformation_context) const {
+  // If any of the composite being used as the basis for insertion, the id being
+  // inserted, or the result id arising from the insertion are irrelevant then
+  // do not add any data synonym facts.  (The result id can be irrelevant if the
+  // insertion occurs in a dead block.)
```

Comment:
If just the id being inserted is irrelevant, could we still add synonym facts about the other components? I suppose the problem is that if any part of a composite is irrelevant, then we have to regard the composite's id as irrelevant.

Reply:
Hmm, you make a good point here - I need to think about this.

Reply:
OK, I reworked this as we discussed.